src/http/mod.rs

```diff
@@ -0,0 +1,75 @@
+//! HTTP transport modules.
+
+#[cfg(feature = "simple_http")]
+pub mod simple_http;
+
+#[cfg(feature = "minreq_http")]
+pub mod minreq_http;
+
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+use std::{fmt, io};
+
+/// The default TCP port to use for connections.
+/// Set to 8332, the default RPC port for bitcoind.
+pub const DEFAULT_PORT: u16 = 8332;
+
+/// The Default SOCKS5 Port to use for proxy connection.
+/// Set to 9050, the default RPC port for tor.
+// Currently only used by `simple_http` module, here for consistency.
+#[cfg(feature = "proxy")]
+pub const DEFAULT_PROXY_PORT: u16 = 9050;
+
+/// Reads a 'user:password' string from `cookie`.
+///
+/// This function exists to support reading the cookie file used by `bitcoind`.
+pub fn read_auth_from_file(cookie: &Path) -> Result<(String, String), CookieError> {
```

Comment:
In 9b566579a2ccfd40d95056342ba45b58f14d09be:

...but then you have this function which isn't compatible with the `cookie_auth` function. I think you could drop this entirely and just use `fs::read_to_string`.